pyrefly/lib/alt/class/class_field.rs

```diff
@@ -1780,6 +1780,116 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
             .is_some_and(|metadata| metadata.fields.contains_key(field_name))
     }
 
+    fn check_typed_dict_override_rules(
```

Comment:
This feels like we duplicated some of the checks from the regular class field override checks - is this necessary, or can we reuse the regular override checks & add on some additional checks just for typed dicts?